File: Day 0016/441. Arranging Coins.java
class Solution {
    public int arrangeCoins(int n) {
        long left = 1, right = 100000;
        int result = 1;

        while (left <= right) {
            int mid = (int) ((left + right) / 2);
            long current = (long) mid * (mid + 1) / 2;

            if (current > n) {
                right = mid - 1;
            } else {
                result = mid;
                left = mid + 1;
            }
        }

        return result;
    }
}

File: Day 0016/441. Arranging Coins.cpp
using ll = long long;

class Solution
{
public:
    int arrangeCoins(int n)
    {
        ll l = 1, r = 1e5;
        int res = 1;
        while (l <= r)
        {
            int mid = (r + l) / 2;
            ll cur = mid * 1LL * (mid + 1) / 2;
            if (cur > (ll)n)
                r = mid - 1;
            else
            {
                res = mid;
                l = mid + 1;
            }
        }
        return res;
    }
};